### ISSUE TITLE
Reproduce more than 1 video in PreRoll

### ISSUE THREAD

**Issue by mirodrigom** [videojs/videojs-contrib-ads#443]
Hello everyone, i want to reproduce more than 1 or 2 videos in preroll.
I could not find any solution in the examples and docs.

Is there any way to do that?

Thank you.

**Comment by incompl** [comment]
The way this is implemented in an ad plugin I work on that uses this project is that the ad plugin plays multiple ads within the same ad break. videojs-contrib-ads doesn't care what happens during the ad break, it just knows it's over when `endLinearAdMode` is called, so multiple ads can be played during that time.

**Comment by nickopris** [comment]
I just needed to implement this for a project and I have a prototype working. I will post here my solution for anyone looking for something basic.

Working with the minimal example file I added a new option to the video element as follows:

`vpreload="ad1.mp4,ad2.mp4" `

Then the plugin looks like this:

`var player = videojs('examplePlayer');
var globalAds = player.options().vpreload.split(","); //get our ads in an array, could be set as array in html as well

player.ads(); // initialize videojs-contrib-ads

// request ads whenever there's new video content
player.on('contentchanged', function () {
  // in a real plugin, you might fetch your ad inventory here
  player.trigger('adsready');
});

player.on('readyforpreroll', function () {
  player.ads.startLinearAdMode();
  playNextAd(globalAds);
});

/**
 * Attach next ad until all ads finished playing.
 * @param globalAds
 */
function playNextAd(globalAds) {
  player.src(globalAds[0]);

  // send event when ad is playing to remove loading spinner
  player.one('adplaying', function () {
    player.controlBar.addClass('vjs-hide');
    player.trigger('ads-ad-started');
  });

  // resume content when all your linear ads have finished
  player.one('adended', function () {

    globalAds.shift();
    if(globalAds.length > 0) {
      playNextAd(globalAds);
      return;
    }

    player.ads.endLinearAdMode();
    player.controlBar.removeClass('vjs-hide');
  });
  
}

player.trigger('adsready');`

The class `vjs-hide` will just hide controls during ads.
